pkg/cvo/status_history.go

```diff
@@ -0,0 +1,235 @@
+package cvo
+
+import (
+	"math"
+	"strings"
+
+	"k8s.io/klog/v2"
+
+	configv1 "github.com/openshift/api/config/v1"
+)
+
+const (
+	// maxFinalEntryIndex is an upper bound index. History entries from 0 to maxFinalEntryIndex are given mostImportantWeight.
+	maxFinalEntryIndex = 4
+
+	// mostImportantWeight is given to history entries that should never be removed including the initial entry,
+	// the most recent entries (indices 0..maxFinalEntryIndex), and the most recently completed update.
+	mostImportantWeight = 1000.0
+
+	// interestingWeight is given to the first or last completed update in a given minor. These are interesting
+	// but not critical.
+	interestingWeight = 30.0
+
+	// partialMinorWeight is given to any partial minor updates between a completed minor version transition.
+	partialMinorWeight = 20.0
+
+	// partialZStreamWeight is given to any partial z-stream updates between a completed z-stream version transition.
+	partialZStreamWeight = -20.0
+
+	// sliceIndexWeight when applied will favor more recent updates and avoid ties.
+	sliceIndexWeight = -1.01
+)
+
+// maxHistorySize is set by the function prune and used by all the history pruning functions. Setting and using
+// this local package variable eases unit testing by allowing testing against a smaller history slice.
+var maxHistorySize int
```

Comment:
I don't understand the need for this package local variable, when `prune` takes an explicit `maxSize` that it can pass through to functions it calls in turn.  Can we drop this?

Reply:
I did this at first thinking there'd be many more functions needing it. But there's only four so sure, I can pass it.